torch_xla/experimental/xla_sharding.py

```diff
@@ -79,6 +81,27 @@ def get_axis_name_idx(self, name: str) -> int:
       return None
     return self.axis_names.index(name)
 
+  def get_op_sharding(self,
```

Comment:
Why don't you use `@functools.lru_cache()` like `global_runtime_device_count`

Reply:
I didn't think about `functools.lru_cache` until handling `global_runtime_device_count`, but I guess I reimplemented it for `get_op_sharding` lol! Let's use it here too.

Reply:
I set `maxsize=None` so that nothing gets evicted.